lib/pbench/server/api/resources/query_apis/iteration_samples.py

```diff
@@ -0,0 +1,279 @@
+from flask import jsonify
+from http import HTTPStatus
+from logging import Logger
+
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.auth import Auth
+from pbench.server.api.resources import (
+    JSON,
+    Schema,
+    Parameter,
+    ParamType,
+    PostprocessError,
+)
+from pbench.server.api.resources.query_apis import CONTEXT, ElasticBase
+from pbench.server.database.models.datasets import Dataset
+from pbench.server.database.models.users import User
+
+
+class IterationSamples(ElasticBase):
+    """
+    Create iteration samples aggregation based on a given run id.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("start", ParamType.DATE, required=True),
+                Parameter("end", ParamType.DATE, required=True),
+                Parameter("run_id", ParamType.STRING, required=True),
```

Comment:
From what you were saying about fio, it sounds like scrolling isn't actually viable, and we should be looking for an "after" point and maybe a "point in time" reference, as well.

Reply:
@webbnh the problem is that for the result-data-sample documents there isn't a clear _after point_ unfortunately. Iteration number which we are using now for sorting can not be used as an after point, since there could be thousands of documents with the same iteration number (kibana showed almost 33,000 documents with the same iteration number for a single run.id). Therefore even if we return 10,000 documents and then use `search_after` using the iteration number we could lose 23000 documents because its gonna skip all those documents. Now even the timestamp is not unique across all the hits of the result-data-sample documents, there are multiple documents with same timestamp and we end-up in same situation where we can't use timestamp for search_after

Reply:
Yeah, we _really_ need to figure out **why** somebody might want _all_ of these documents.

That is, if they want a particular time-series, then we can select that and page it with `search_after` using the timestamp.  (That is, I'm not surprised that timestamps aren't unique across different sets of time-series data, but I bet within a single series they are unique.)  Alternatively, we can have ES do time-sampled aggregations of the time-series data, and return however many "buckets" as they want (within reason...)...which will get us down from tens of thousands to whatever number the server and the dashboard agree on.

But, if they aren't actually interested in the time-series data, then I bet we can have ES aggregate all the documents for us, and we can return a single result.  I seem to recall that what the dashboard really wanted was a set of column headings that it could offer to the user; I'm guessing that the user's choice would then be used to select which actual documents to fetch.  So, I'm guessing that there's a solution here which doesn't involve returning all XX,000 documents on the first pass (or ever).